update-agent-loader/sign.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+from cryptography.hazmat.primitives.asymmetric import ed25519
+
+WLD_TAG = b"$WLD TO THE MOON"
+TAG_LEN = len(WLD_TAG)
+SIG_LEN_SIZE = 4  # 4 bytes for signature length
+
+def load_raw_ed25519_private_key(path):
+    with open(path, "rb") as f:
+        raw_key = f.read()
+    if len(raw_key) != 32:
+        raise ValueError("Expected 32-byte Ed25519 private key")
+    return ed25519.Ed25519PrivateKey.from_private_bytes(raw_key)
+
+def read_and_strip_existing_signature(data):
+    """If data ends with WLD_TAG and signature length, strip them."""
+    if len(data) < TAG_LEN + SIG_LEN_SIZE:
+        return data, False
+
+    if data[-TAG_LEN:] != WLD_TAG:
```

Comment:
This sort of manual binary parsing is the type of spot that bugs get introduced in security critical code.

Id eliminate the magic string, stick to known key formats, and use rust not python.